etc/kayobe/inventory/group_vars/all/firewall

```diff
@@ -0,0 +1,437 @@
+    selectattr('enabled', 'true') |
+    map(attribute='rules') |
+    flatten |
+    selectattr('network', 'in', network_interfaces) |
+    selectattr('zone') |
+    union(stackhpc_compute_firewalld_rules_extra) |
+    unique |
+    select }}
+
+stackhpc_compute_firewalld_rules_default:
+  # Common
+  - rules:
+      - service: ssh
+        zone: "{{ admin_oc_net_name | net_zone }}"
+        network: "{{ admin_oc_net_name }}"
+        state: enabled
+      - service: dhcpv6-client
+        zone: "{{ public_net_name | net_zone }}"
+        network: "{{ public_net_name }}"
+        state: disabled
+    enabled: true
+  - rules:
+      - service: ssh
+        state: disabled
+        zone: "{{ public_net_name | net_zone }}"
+        network: "{{ public_net_name }}"
+    enabled: "{{ public_net_name | net_zone != provision_oc_net_name | net_zone }}"
+  # GENEVE
+  - rules:
+      - port: 6081/udp
+        state: enabled
+        zone: "{{ tunnel_net_name | net_zone }}"
+        network: "{{ tunnel_net_name }}"
+    enabled: "{{ ('geneve' in (kolla_neutron_ml2_type_drivers + kolla_neutron_ml2_tenant_network_types)) | bool }}"
+  # VXLAN
+  - rules:
+      - port: 4789/udp
+        state: enabled
+        zone: "{{ tunnel_net_name | net_zone }}"
+        network: "{{ tunnel_net_name }}"
+    enabled: "{{ ('vxlan' in (kolla_neutron_ml2_type_drivers + kolla_neutron_ml2_tenant_network_types)) | bool }}"
+
+stackhpc_compute_firewalld_rules_extra: []
+
+###############################################################################
+# Storage firewalld rules
+
+stackhpc_storage_firewalld_rules: >-
+  {{ stackhpc_storage_firewalld_rules_default |
+    selectattr('enabled', 'true') |
+    map(attribute='rules') |
+    flatten |
+    selectattr('network', 'in', network_interfaces) |
+    selectattr('zone') |
+    union(stackhpc_storage_firewalld_rules_extra) |
+    unique |
+    select }}
+
+stackhpc_storage_firewalld_rules_default:
+  # Common
+  - rules:
+      - service: ssh
+        zone: "{{ admin_oc_net_name | net_zone }}"
+        network: "{{ admin_oc_net_name }}"
+        state: enabled
+      - port: 9283/tcp
+        zone: "{{ provision_oc_net_name | net_zone }}"
+        network: "{{ provision_oc_net_name }}"
+        state: enabled
+    enabled: true
+  - rules:
+      - service: ssh
+        zone: "{{ storage_net_name | net_zone }}"
+        network: "{{ storage_net_name }}"
+        state: disabled
+    enabled: "{{ storage_net_name | net_zone != provision_oc_net_name | net_zone }}"
+  # Ceph
+  - rules:
+      - service: ceph
+        zone: "{{ storage_net_name | net_zone }}"
+        network: "{{ storage_net_name }}"
+        state: enabled
+      - service: ceph-mon
+        zone: "{{ storage_net_name | net_zone }}"
+        network: "{{ storage_net_name }}"
+        state: "{{ 'enabled' if 'mons' in group_names else 'disabled' }}"
+    enabled: "{{ stackhpc_enable_ceph | default(false) | bool }}"
```

Comment:
I'd just use the inventory here rather than enabling a new flag. There is a `ceph` group.